cpp/src/parquet/properties.h

```diff
@@ -176,59 +176,100 @@ class PARQUET_EXPORT WriterProperties {
       return this;
     }
 
+    /**
+     * Enable dictionary encoding in general for all columns. Default enabled.
+     */
     Builder* enable_dictionary() {
       default_column_properties_.set_dictionary_enabled(true);
       return this;
     }
 
+    /**
+     * Disable dictionary encoding in general for all columns. Default enabled.
+     */
     Builder* disable_dictionary() {
       default_column_properties_.set_dictionary_enabled(false);
       return this;
     }
 
+    /**
+     * Enable dictionary encoding for column specified by `path`. Default enabled.
+     */
     Builder* enable_dictionary(const std::string& path) {
       dictionary_enabled_[path] = true;
       return this;
     }
 
+    /**
+     * Enable dictionary encoding for column specified by `path`. Default enabled.
+     */
     Builder* enable_dictionary(const std::shared_ptr<schema::ColumnPath>& path) {
       return this->enable_dictionary(path->ToDotString());
     }
 
+    /**
+     * Disable dictionary encoding for column specified by `path`. Default enabled.
+     */
     Builder* disable_dictionary(const std::string& path) {
       dictionary_enabled_[path] = false;
       return this;
     }
 
+    /**
+     * Disable dictionary encoding for column specified by `path`. Default enabled.
+     */
     Builder* disable_dictionary(const std::shared_ptr<schema::ColumnPath>& path) {
       return this->disable_dictionary(path->ToDotString());
     }
 
+    /**
+     * Specify the dictionary page size limit per row group. Default 1MB.
+     */
     Builder* dictionary_pagesize_limit(int64_t dictionary_psize_limit) {
       dictionary_pagesize_limit_ = dictionary_psize_limit;
       return this;
     }
 
+    /**
+     * Specify the write batch size while writing batches of Arrow values into Parquet.
+     * Default 1024.
+     */
     Builder* write_batch_size(int64_t write_batch_size) {
       write_batch_size_ = write_batch_size;
       return this;
     }
 
+    /**
+     * Specify the max row group length.
+     * Default 64M.
```

Comment:
Perhaps this should be changed to 1M, as discussed in user group a month ago with @westonpace